code/modules/mod/modules/modules_supply.dm

```diff
@@ -0,0 +1,577 @@
+	var/image/charge_up_overlay
+
+/obj/item/mod/module/hydraulic/Initialize(mapload)
+	. = ..()
+	charge_up_overlay = image(icon = 'icons/effects/effects.dmi', icon_state = "electricity3", layer = EFFECTS_LAYER)
+
+/obj/item/mod/module/hydraulic/on_select_use(atom/target)
+	. = ..()
+	if(!.)
+		return
+	var/current_time = world.time
+	var/atom/movable/plane_master_controller/pm_controller = mod.wearer.hud_used.plane_master_controllers[PLANE_MASTERS_GAME]
+	for(var/key in pm_controller.controlled_planes)
+		animate(pm_controller.controlled_planes[key], launch_time, transform = matrix(1.25, MATRIX_SCALE))
+	mod.wearer.visible_message("<span class='warning'>[mod.wearer] starts whirring!</span>")
+	playsound(src, 'sound/items/modsuit/loader_charge.ogg', 75, TRUE)
+	mod.wearer.add_overlay(charge_up_overlay)
+	var/power = launch_time
+	if(!do_after(mod.wearer, launch_time, target = mod.wearer))
+		power = world.time - current_time
+	drain_power(use_power_cost)
+	for(var/key in pm_controller.controlled_planes)
+		animate(pm_controller.controlled_planes[key], 0.1 SECONDS, transform = matrix(1, MATRIX_SCALE))
+	playsound(src, 'sound/items/modsuit/loader_launch.ogg', 75, TRUE)
+	var/angle = get_angle(mod.wearer, target)
+	mod.wearer.transform = mod.wearer.transform.Turn(angle)
+	mod.wearer.throw_at(get_ranged_target_turf_direct(mod.wearer, target, power), \
+		range = power, speed = max(round(0.2*power), 1), thrower = mod.wearer, spin = FALSE, \
+		callback = CALLBACK(src, PROC_REF(on_throw_end), mod.wearer, -angle))
+
+/obj/item/mod/module/hydraulic/proc/on_throw_end(mob/user, angle)
+	if(!user)
+		return
+	user.transform = user.transform.Turn(angle)
+	user.cut_overlay(charge_up_overlay)
+
+/obj/item/mod/module/disposal_connector
+	name = "MOD disposal selector module"
+	desc = "A module that connects to the disposal pipeline, causing the user to go into their config selected disposal. \
+		Only seems to work when the suit is on."
+	icon_state = "disposal"
+	complexity = 2
+	idle_power_cost = DEFAULT_CHARGE_DRAIN * 0.3
+	incompatible_modules = list(/obj/item/mod/module/disposal_connector)
+	var/disposal_tag = NONE
+
+/obj/item/mod/module/disposal_connector/Initialize(mapload)
+	. = ..()
+	disposal_tag = pick(GLOB.TAGGERLOCATIONS)
+
+/obj/item/mod/module/disposal_connector/on_suit_activation()
+	RegisterSignal(mod.wearer, COMSIG_MOVABLE_DISPOSING, PROC_REF(disposal_handling))
+
+/obj/item/mod/module/disposal_connector/on_suit_deactivation(deleting = FALSE)
+	UnregisterSignal(mod.wearer, COMSIG_MOVABLE_DISPOSING)
+
+/obj/item/mod/module/disposal_connector/get_configuration()
+	. = ..()
+	.["disposal_tag"] = add_ui_configuration("Disposal Tag", "list", GLOB.TAGGERLOCATIONS[disposal_tag], GLOB.TAGGERLOCATIONS)
+
+/obj/item/mod/module/disposal_connector/configure_edit(key, value)
+	switch(key)
+		if("disposal_tag")
+			for(var/tag in 1 to length(GLOB.TAGGERLOCATIONS))
+				if(GLOB.TAGGERLOCATIONS[tag] == value)
+					disposal_tag = tag
+					break
```

Comment:
double check that this is the proper way to do this with the disposals tagger (go look at their TGUI and see how it does it)

Reply:
will do

Reply:
Removing it for now, due to how destinations are handled / it's a "bepis module" which basically means it is supposed to be rare from tg, may port it in a future one,